### PR TITLE
Custom Mask support for Textbox

### PR DESCRIPTION
**Pull request by Darrken**
- Added a customMask prop and new "custom" type to maskType prop.
- Removed a few props that don't require explicit definition (shouldn't break any prior implementation).
- Updated docs with better explanations for additional props.

### CODE REVIEW

File: packages/es-components/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "es-components",
-  "version": "15.1.1",
+  "version": "15.2.0"
```
Comment (aabenoja) [inline packages/es-components/package.json]: How'd the comma get removed?

**Comment by aabenoja** [comment]
@Darrken Is this PR still relevant?

**Comment by Darrken** [comment]
Team 5 was having some issues testing one of the inputs with a text mask due to the `keepCharPositions` being on. Even though fixing the tests is the preferred solution, I thought it might be useful to add the ability to provide your own custom mask if needed. This also gets rid of some extraneous props.

If you guys don't think it's useful, we can ditch it though.

**Comment by aabenoja** [comment]
It's not that I didn't think this is useful, but rather I see [this commit](https://github.com/WTW-IM/es-components/commit/a606f522d03ad2c3b0ca6b4465f1a5cd836d335c) in master and this looks exactly the same as this PR. At the very least, I'm having an issue seeing the different between that commit and this PR.

**Comment by Darrken** [comment]
Ah, I think Jake maybe pulled that in while he was coalescing some other PRs a couple of weeks ago. I didn't notice. I'll close this out.

**Pull request closed without merging**